Authenticator/Source/AppViewController.swift

```diff
@@ -86,17 +99,40 @@ extension AppViewController: AppPresenter {
             let scannerViewController = TokenScannerViewController(actionHandler: self)
             presentViewController(scannerViewController)
 
-        case .EntryForm(let form):
-            let formController = TokenFormViewController(viewModel: form.viewModel,
-                actionHandler: form)
-            form.presenter = formController
-            presentViewController(formController)
-
-        case .EditForm(let form):
-            let editController = TokenFormViewController(viewModel: form.viewModel,
-                actionHandler: form)
-            form.presenter = editController
-            presentViewController(editController)
+        case .EntryForm(let formViewModel):
+            if case .EntryForm = currentViewModel.modal,
+                let editController = modalNavController?.topViewController as? TokenFormViewController {
+                editController.updateWithViewModel(formViewModel)
+            } else {
+                let formController = TokenFormViewController(viewModel: formViewModel,
+                    actionHandler: entryFormActionHandler)
+                presentViewController(formController)
+            }
+
+        case .EditForm(let formViewModel):
+            if case .EditForm = currentViewModel.modal,
+                let editController = modalNavController?.topViewController as? TokenFormViewController {
```

Comment:
Line Length Violation: Line should be 100 characters or less: currently 104 characters (line_length)